### PR TITLE
Flattening out the RDS explorer nodes

### PR DESCRIPTION
**Pull request by kiiadi**
Previously:
Screen Shot 2020-10-14 at 3 50 21 PM

Now:
Screen Shot 2020-10-14 at 2 25 01 PM

### CODE REVIEW

File: jetbrains-ultimate/src-201+/software/aws/toolkits/jetbrains/services/rds/RdsEngine.kt [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import software.aws.toolkits.resources.message
+import javax.swing.Icon
+
+enum class RdsEngine(val engine: String, val displayName: String, val icon: Icon, val resourceType: String = engine) {
+    MySql("mysql", message("rds.mysql"), AwsIcons.Resources.Rds.MYSQL) {
+        override fun iamConnectionStringUrl(url: String) = "jdbc:$jdbcMysql://$url/"
+    },
+    Postgres("postgres", message("rds.postgres"), AwsIcons.Resources.Rds.POSTGRES) {
+        override fun iamConnectionStringUrl(url: String) = "jdbc:$jdbcPostgres://$url/"
+        override fun iamUsernameHook(username: String) = username.toLowerCase()
```
Comment (hunterwerlla) [inline jetbrains-ultimate/src-201+/software/aws/toolkits/jetbrains/services/rds/RdsEngine.kt:16]: Can you copy the comment back to here?

File: jetbrains-ultimate/src-201+/software/aws/toolkits/jetbrains/services/rds/RdsEngine.kt [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    };
+
+    abstract fun iamConnectionStringUrl(url: String): String
+
+    /**
+     * Some engines need to manipulate the username (e.g. Postgres should be lower-cased)
+     */
+    open fun iamUsernameHook(username: String) = username
```
Comment (hunterwerlla) [inline jetbrains-ultimate/src-201+/software/aws/toolkits/jetbrains/services/rds/RdsEngine.kt:33]: can we drop the `Hook` at the end?

**Review by hunterwerlla — COMMENTED** [review]
Do the actions work after this?

Also, a thought, maybe we should only use the `[]` to mark it as `Aurora`, the icon should be enough?

File: jetbrains-ultimate/src-201+/software/aws/toolkits/jetbrains/services/rds/RdsResources.kt [hunk 3]
```diff
@@ -35,16 +31,10 @@ object RdsResources {
...
-    val LIST_INSTANCES_MYSQL: Resource.Cached<List<DBInstance>> = listInstancesFilter(mysqlEngineType)
-    val LIST_INSTANCES_POSTGRES: Resource.Cached<List<DBInstance>> = listInstancesFilter(postgresEngineType)
-    val LIST_INSTANCES_AURORA_MYSQL: Resource.Cached<List<DBInstance>> = listInstancesFilter(auroraMysqlEngineType)
-    val LIST_INSTANCES_AURORA_POSTGRES: Resource.Cached<List<DBInstance>> = listInstancesFilter(auroraPostgresEngineType)
 
-    private fun listInstancesFilter(engine: String) = ClientBackedCachedResource(RdsClient::class, "rds.list_instances.$engine") {
-        this.describeDBInstancesPaginator {
-            it.filters(Filter.builder().name(engineFilter).values(engine).build())
-        }.toList().flatMap { it.dbInstances() }
+    val LIST_SUPPORTED_INSTANCES: Resource.Cached<List<DBInstance>> = ClientBackedCachedResource(RdsClient::class, "rds.list_supported_instances") {
```
Comment (abrooksv) [inline jetbrains-ultimate/src-201+/software/aws/toolkits/jetbrains/services/rds/RdsResources.kt:35]: Guess this fixes the issue with 5.7 reported in #2124 (in respect to the explorer)

Comment (abrooksv) [reply jetbrains-ultimate/src-201+/software/aws/toolkits/jetbrains/services/rds/RdsResources.kt:35]: Nvm, no it doesn't. Aurora Mysql has 2 engine values now, so we need to cope with that.

File: jetbrains-ultimate/src-201+/software/aws/toolkits/jetbrains/services/rds/RdsEngine.kt (hunk 1 above)
Comment (kiiadi) [reply jetbrains-ultimate/src-201+/software/aws/toolkits/jetbrains/services/rds/RdsEngine.kt]: > Also, a thought, maybe we should only use the [] to mark it as Aurora, the icon should be enough?

So you're saying 'plain' DB's don't have a status, but the aurora ones do? I can get on board with that.

Comment (hunterwerlla) [reply jetbrains-ultimate/src-201+/software/aws/toolkits/jetbrains/services/rds/RdsEngine.kt]: Yep, exactly

**Comment by kiiadi** [comment]
Screen Shot 2020-10-15 at 1 34 51 PM
Applied suggestion to only show the additional information for Aurora DBs.

File: jetbrains-ultimate/src-201+/software/aws/toolkits/jetbrains/services/rds/RdsEngine.kt (hunk 1 above)
Comment (kiiadi) [reply jetbrains-ultimate/src-201+/software/aws/toolkits/jetbrains/services/rds/RdsEngine.kt]: done

**Review by hunterwerlla — APPROVED** [review]

**Pull request merged**